### PR TITLE
Fix Incorrect Imports Following the Type System Changes

### PR DESCRIPTION
**Pull request by Convly**
### What does it do?

- Fix generators so that they use the new Strapi types imports
- Fix old imports in the code of our packages

### Why is it needed?

#19656 has been merged, but we still need to adapt the templates and generators so that they output the correct types.

On top of this, we still have code using the old imports, which was surprisingly not caught by TS.

### Related issue(s)/PR(s)

see #19656 for more information about the type changes

### CODE REVIEW

File: packages/generators/app/src/resources/files/ts/src/admin/app.example.tsx [hunk 1]
```diff
@@ -29,7 +29,7 @@ export default {
       // 'zh',
     ],
   },
-  bootstrap(app) {
+  bootstrap(app: unknown) {
```
Comment (innerdvations) [inline packages/generators/app/src/resources/files/ts/src/admin/app.example.tsx:32]: Isn't this the front-end Strapi app object? I think there's a type for it somewhere

**Review by joshuaellis — APPROVED** [review]
You Cant See Me John Cena GIF by WWE

File: packages/generators/app/src/resources/files/ts/src/admin/app.example.tsx (hunk 1 above)
Comment (Convly) [reply packages/generators/app/src/resources/files/ts/src/admin/app.example.tsx:32]: @joshuaellis does it ring a bell?

Comment (joshuaellis) [reply packages/generators/app/src/resources/files/ts/src/admin/app.example.tsx:32]: There's not one exported, but actually, why dont we do that now? 

if we add:

```ts
export type { StrapiApp } from './StrapiApp';	
```

to the strapi-admin package then you can import it via:

```ts
export type { StrapiApp } from '@strapi/strapi/admin'
```

Comment (joshuaellis) [reply packages/generators/app/src/resources/files/ts/src/admin/app.example.tsx:32]: Then it'll be typed!

Comment (Convly) [reply packages/generators/app/src/resources/files/ts/src/admin/app.example.tsx]: Done, thanks :magic_wand:

Comment (innerdvations) [reply packages/generators/app/src/resources/files/ts/src/admin/app.example.tsx]: beautiful!

**Review by innerdvations — COMMENTED** [review]
There's a failing front-end test I don't understand at all, but other than that all good!

**Comment by joshuaellis** [comment]
> There's a failing front-end test I don't understand at all, but other than that all good!

that's a flakey test with an async thing not being awaited, i think Mark K is looking at it. re-running _should_ work.

**Review by innerdvations — APPROVED** [review]

**Pull request merged**